src/tool/cppwinrt/strings/base_coroutine.h

```diff
@@ -0,0 +1,125 @@
+
+#ifndef __clang__
+
+#include <experimental/coroutine>
+
+#else
+
+namespace std::experimental
```

Comment:
Since there's a compiler difference that you're smoothing over, it seems likely that projects could accidentally take a dependency on your adapter layer. Would it make sense to put the adapter in a cppwinrt::details namespace, then in the VC version, just do namespace cppwinrt::details { using std::experimental; }, then consume from there so that clang projects using cppwinrt can't easily take an accidental dependency on something that you might want to change later once things standardize a bit more?

Reply:
The trick is that everything here is part of the public library support for coroutines. This is intentionally designed for libraries to take a dependency on and is unavoidable. As such, care is taken that this accurately represents the same library surface as the VC header and the C++20 standard - specifically nothing *more* is provided. The only implementation detail is the implementation of the functions that call the Clang intrinsics.